### ISSUE TITLE
Confusing priority of vault_url value sources

### ISSUE THREAD

**Issue by kazauwa** [nymous/pydantic-vault#17]
I encountered the problem that the `vault_url` value specified in `class Config` takes priority over the `VAULT_ADDR` environment variable. In my head, it should be the opposite: we define a default value in the code, but we can override it externally by setting an environment variable (see [12-factor app](https://12factor.net/config)). Although, I found out that this can be achieved by setting `vault_url` to `None`, I still find this confusing.

Luckily, it can be easily fixed by rearranging the order of reading vault URL sources in `_get_authenticated_vault_client()`:

```python3
_vault_url: Optional[str] = None
if getattr(settings.__config__, "vault_url", None) is not None:
    _vault_url = settings.__config__.vault_url  # type: ignore
    logger.debug(f"Found Vault Address '{_vault_url}' in Config")
if "VAULT_ADDR" in os.environ:
    _vault_url = os.environ["VAULT_ADDR"]
    logger.debug(f"Found Vault Address '{_vault_url}' in environment variables")
if _vault_url is None:
    raise VaultParameterError("No URL provided to connect to Vault")
```

I can make a pull request with the change and cover them with tests if that's okay with you.

**Comment by vladimir-kotikov** [comment]
If we're going to refer to pydantic's list of priorities then I personally would say that defining `vault_url` in `Config` class is more like [number 5](https://docs.pydantic.dev/1.10/usage/settings/#field-value-priority) because it essentialy bakes the parameter into class definition and hence need to have the lowest priority.

So shortly speaking I basically agree with Igor - that's going to be a good candidate to be changed.

**Comment by nymous** [comment]
I agree that it does make sense.

Also by giving the link to Pydantic docs I didn't want to say "this is how we should do things", I was just trying to remember where my decision was coming from :sweat_smile:

**Comment by ingvaldlorentzen** [comment]
I agree that environment variables should override what is written in the `Config` class. 
We've "implemented" this ourselves with: 
`vault_address = os.environ.get('VAULT_ADDR', 'https://default-vault.example')`

**Comment by kazauwa** [comment]
Hi everyone!
Thanks for participating! I'm glad we're on the same page with this. I'm still up for the task if you're happy with the change, I could do it over this weekend.

**Comment by nymous** [comment]
Thanks everyone! I am now convinced that this should change, for all environment variables used in the `Config` class.
@kazauwa, I would gladly accept your contribution ^^ I wanted to release #24 first because I knew it would add a new env variable, so I didn't reply earlier. But this would mark the "great 1.0 release"! :tada:

**Comment by nymous** [comment]
This is now released in 1.0.0, thank you again @kazauwa! :rocket: :fireworks:
